### ISSUE TITLE
[Issue]: It applies changes forever.

### ISSUE THREAD

**Issue by laichiaheng** [an-anime-team/an-anime-game-launcher#50]
### Distro

Arch Linux

### Installation Type

AUR

### Describe the issue

It has been stuck here for a long time.
![螢幕快照 2023-01-18 20-45-54](https://user-images.githubusercontent.com/9532105/213175108-7e66f252-0a01-4226-b40f-caa36788f21b.png)

**Comment by lubuoren** [comment]
I think the problem is about the binary _hdiffz_ and _hpatchz_.I think you can remove all the chosen of voice pack and install the voice pack in game.(I don't know if it is right,because I can't apply the patch of anti-cheat)

**Comment by c02y** [comment]
I tried several times, it gets stuck on the `Applying changes... 100%` after pressing `Apply changes` button, restart and do it again, it won't get into the next step.

![image](https://user-images.githubusercontent.com/2135996/213162840-13cd22a1-8857-483d-86c6-3d6e7d4cb762.png)


The latest.log file contains hundreds of lines like:
```
[thread: 5535][Launcher/State/ApplyChanges] Patching GenshinImpact_Data/StreamingAssets/Audio/GeneratedSoundBanks/Windows/English(US)/1015.pck: failure
```

It seems the `Update` button will download some file `en-us_3.3.0_3.4.0_hdiff_kxxx.zip` and then `Apply changes` will unzip it and fail in the end.

![image](https://user-images.githubusercontent.com/2135996/213165587-bc088ebc-0b4e-4283-b9f7-be512871d1a2.png)

**Comment by driccio98** [comment]
in my case is not failing to unzip the file as is not even present when the error occurs
![image](https://user-images.githubusercontent.com/57051444/213183492-dda52d34-a9f9-489c-8983-5ec09b3e2a37.png)

**Comment by c02y** [comment]
@driccio98 

The zip file is deleted automaitcally after getting unzipped, I quickly took the screenshot before it is deleted.

**Comment by driccio98** [comment]
Tried updating hpatchz to the last version from [here](https://github.com/sisong/HDiffPatch), same result.

**Comment by driccio98** [comment]
When I try to manually run the hpatchz binary to patch a .pck as the launcher is trying to do, I get this error 
![image](https://user-images.githubusercontent.com/57051444/213189294-8b198ed0-5c33-4fe1-9587-62d0bfdcbcde.png)
Seems like the .hdiff file contains wrong info about the size off the .pck file and is failing to patch it.

**Comment by zeff9801** [comment]
i have the same issue, haven't been able to fix it yet

**Comment by driccio98** [comment]
I tried downloading all the updates directly and patching them myself, I got the same result, this makes me think that maybe the problem are the hdiff files themselves. Windows users are able to manually patch the files for 3.4 in the same method the launcher patches them.

**Comment by germaniuss** [comment]
I had the same problem and I don't know what the difference is but the gtk version downloads and patches fine.

**Comment by scottdraper8** [comment]
I'm experiencing the same issue with the AUR installation on Manjaro. I even wiped all existing files and re-downloaded everything, in case the problem occurred only when updating a previous version. Still running into the same issue. I'll give the GTK version a try

**Comment by Ajhayter** [comment]
Same symptoms on SteamOS (deck). Something is stuck but I'm not savvy enough to ID what it is; Hopefully as described above.

**Comment by driccio98** [comment]
A workaround is to use the gtk launcher, is working normally as of now

**Comment by warrinerbros** [comment]
> o

Here's the link to that: https://github.com/an-anime-team/an-anime-game-launcher-gtk

**Comment by tigersoldier** [comment]
You need to manually fix the in-game data messed up by the voice update process. Open the game folder in settings. Remove the following files:
 * hdifffiles.txt
 * GenshinImpact_Data/StreamingAssets/Audio/GeneratedSoundBanks/Windows/English/.version  (this file is hidden by default)
 * (also the .version file in other languages if you have any)

**Comment by Ajhayter** [comment]
> You need to manually fix....

Can confirm this allowed the install to compete on SteamOS. 
Patch applied, game launched, now verifying files. Thanks :)

**Comment by Mar0xy** [comment]
Should be fixed in 2.4.4 release
